sklearn/manifold/_t_sne.py

```diff
@@ -681,12 +688,30 @@ def __init__(self, n_components=2, *, perplexity=30.0,
     def _fit(self, X, skip_num_points=0):
         """Private function to fit the model using X as training data."""
 
+        if self.init == 'warn':
+            # See issue #18018
+            warnings.warn("The default initialization will change from "
+                          "random to PCA in 1.01.", FutureWarning)
+            self.init = 'random'
+        if self.learning_rate == 'warn':
+            # See issue #18018
+            warnings.warn("The default learning rate will change from "
+                          "200.0 to 'auto' in 1.01.", FutureWarning)
+            self.learning_rate = 200.0
         if self.method not in ['barnes_hut', 'exact']:
             raise ValueError("'method' must be 'barnes_hut' or 'exact'")
         if self.angle < 0.0 or self.angle > 1.0:
             raise ValueError("'angle' must be between 0.0 - 1.0")
         if self.square_distances not in [True, 'legacy']:
             raise ValueError("'square_distances' must be True or 'legacy'.")
+        if self.learning_rate == 'auto':
+            # See issue #18018
+            self.learning_rate = X.shape[0] / self.early_exaggeration / 4
+            self.learning_rate = np.max(self.learning_rate, 50)
```

Comment:
Is 50 better than the previous value of 200 ?

Reply:
I haven't extensively tested it (note that this only matters for relatively small sample sizes, otherwise N/48 heuristic will kick in). 50 corresponds to the default in other implementations including the "official" Barnes Hut implementation of Maaten (where it's 200 but without the factor of 4). Which is why openTSNE uses `max(50, N/12)` rule. But we can ask well keep 200 here for better consistency with the current sklearn default. I have no preference.

Reply:
Ok makes sense, 50 is fine.